.github/workflows/build-push-docker-images.yaml

```diff
@@ -0,0 +1,171 @@
+name: Build-Push-Docker-Image
+
+on:
+  push:
+    branches:
+      - build-push-docker-images
+
+jobs:
+  build-push-base-rvt-image:
+
+    name: Build and push RVT base Docker image
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push rvt_base
+        uses: docker/build-push-action@v2.2.2
+        with:
+          context: docker/base
+          push: true
+          build-args: |
+            USERNAME=rvt
+            USER_UID=1000
+            USER_GID=1000
+            UBUNTU_VERSION=20.04
+          cache-from: type=registry,ref=thierrymarianne/contrib-rvt_base:latest
+          cache-to: type=inline
+          tags: |
+            thierrymarianne/contrib-rvt_base:latest
+
+  build-push-rvt-images:
+
+    needs: build-push-base-rvt-image
+
+    name: Build and push RVT solvers, Klee, SeaHorn, Rust2calltree images
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        registry_namespace: ["thierrymarianne/"]
```

Comment:
Since this repo has a liberal license, it is fine for you to do this in a fork but, unfortunately, we cannot merge this line into our repo because:

1) Any namespace would have to be a Google-owned namespace instead of "thierrymarianne/"
2) Even if we had a google-owned namespace at docker hub, I am told that the right way for a google project to do this is to publish the image in the google-cloud-container-registry
3) Publishing the image in the google-cloud-container-registry is hard for us to do for <reasons>.

However, this looks like a useful change for anybody else to have so maybe I could leave the issue open so that people can find this change and decide whether it is right for them?

Reply:
I was counting on seeing the namespace updated and I understand this might not be trivial enough to be merged.